calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -95,19 +96,19 @@ def _configure_interface(self):
         return ep
 
     def _get_node_ip(self):
-        """Determine the IP for the host node.
-
-        This hits a well-known IP (the Google public DNS).
+        """ 
+            Determine the IP for the host node.
         """
-        # We'd like to use the k8s API here, but it doesn't work in the Vagrant
-        # case since the node's name is set to its IP (so we have no way of
-        # getting from hostname=>IP).
-        # TODO: do this more reliably by parsing 'ip
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        s.connect(('8.8.8.8', 80))
-        ip = s.getsockname()[0]
-        s.close()
-        return ip
+        # Compile list of IPv4 addresses on network, return the first entry
+        ipv4_addresses = get_host_ips(version=4)
+        
+        # If no IPv4 addresses exist, check IPv6, return the first entry
+        if not ipv4_addresses:
+            return get_host_ips(version=6)[0]
```

Comment:
It's a bit dangerous to blindly access an index of a list when we can't guarentee that there's a value there. We probably want to wrap this in a `try/except IndexError` block.

However, the kubernetes plugin doesn't have the best error handling currently - I'm not sure what the best course of action would be in the except statement. Maybe `print("Couldn't determine the host's IP")` and then `sys.exit(1)`, just like we do in calico-docker? I'll defer to @caseydavenport for how we might want to handle this.